src/main/java/cmu/pasta/mu2/diff/guidance/OptimizedMutationGuidance.java

```diff
@@ -3,5 +3,5 @@
 import cmu.pasta.mu2.util.ArraySet;
 
 public interface OptimizedMutationGuidance extends DiffGuidance {
-    ArraySet filterMutants(ArraySet runMutants);
+    ArraySet filterMutants();
```

Comment:
I am not sure I understand this API. Either let's keep it functional so that it has a specific input parameter and return value, or if we want to assume that the filter has access to the guidance's `runMutants` structure then we probably should just have a `void filterMutants()` that updates the structure in place. Does that make sense? If I missed something, let me know why we don't need a parameter coming in but need a return value coming out.